src/assembly/stack.ts

```diff
@@ -0,0 +1,162 @@
+    public readonly destination: FileDestination;
+
+    /**
+     * The destination id
+     */
+    public readonly id: DestinationIdentifier;
+
+    /**
+     * Absolute path to the asset
+     */
+    public readonly path: string;
+    public readonly packaging: FileAssetPackaging;
+
+    /**
+     * @param directory - The directory in which the file manifest is found
+     * @param asset - The file asset
+     */
+    constructor(directory: string, asset: FileManifestEntry) {
+        this.destination = asset.destination;
+        this.id = asset.id;
+        if (asset.source.executable) {
+            throw new Error(`file assets produced by commands are not yet supported`);
+        }
+        this.path = path.join(directory, asset.source.path!);
+        this.packaging = asset.source.packaging ?? FileAssetPackaging.FILE;
+    }
+}
+
+/**
+ * StackManifest represents a single Stack that needs to be converted
+ * It contains all the necessary information for this library to fully convert
+ * the resources and assets in the stack to pulumi resources
+ */
+export class StackManifest {
+    /**
+     * The artifactId / stackId of the stack
+     */
+    public id: string;
+
+    /**
+     * The construct tree for the stack
+     */
+    public readonly constructTree: ConstructTree;
+
+    /**
+     * The relative path to the stack template file
+     */
+    public readonly templatePath: string;
+
+    /**
+     * The Outputs from the CFN Stack
+     */
+    public readonly outputs?: { [id: string]: any };
+
+    /**
+     * The Parameters from the CFN Stack
+     */
+    public readonly parameters?: { [id: string]: CloudFormationParameter };
+
+    /**
+     * Map of resource logicalId to CloudFormation template resource fragment
+     */
+    private readonly resources: { [logicalId: string]: CloudFormationResource };
+
+    /**
+     *
+     */
+    private readonly metadata: StackMetadata;
+    private readonly assets: StackAsset[];
+    private readonly directory: string;
+    constructor(
+        directory: string,
+        id: string,
+        templatePath: string,
+        metadata: StackMetadata,
+        tree: ConstructTree,
+        template: CloudFormationTemplate,
+        assets: StackAsset[],
+    ) {
+        this.directory = directory;
+        this.assets = assets;
+        this.outputs = template.Outputs;
+        this.parameters = template.Parameters;
+        this.metadata = metadata;
+        this.templatePath = templatePath;
+        this.id = id;
+        this.constructTree = tree;
+        if (!template.Resources) {
+            throw new Error('CloudFormation template has no resources!');
+        }
+        this.resources = template.Resources;
+    }
+
+    public get fileAssets(): FileAssetManifest[] {
+        return this.assets
+            .filter((asset) => asset.type === 'file')
+            .flatMap((asset) => new FileAssetManifest(this.directory, asset));
+    }
+
+    // TODO: implement docker assets
```

Comment:
Do we have a ticket for this?

Reply:
Yep! https://github.com/pulumi/pulumi-cdk/issues/125